js/app.jsx

```diff
@@ -159,6 +187,106 @@ var SearchableRemoteListMixin = {
 
 }
 
+var ListToolbar = React.createClass({
+  render: function() {
+    var newButton = <div></div>;
+    var diffButton = <div></div>;
+
+    var diffURL = "";
+    var diffDisabled = true;
+
+    //create diff button
+    if (this.props.setting.pagename =="caseversion") {
+      if (typeof this.props.checked !== "undefined"){
+        diffURL = "diff.html?lhs=" + this.props.checked[0] + "&rhs=" + this.props.checked[1]
+        if (this.props.checked.length == 2){
+          var diffDisabled = false;
+        }
+      }
+      diffButton = <Button bsStyle="success" target="blank_" href={diffURL} disabled={diffDisabled}>diff</Button>
+    }
+
+    //create new button
+    if (this.props.setting.pagename == "caseversion") {
+      newButton = <Button href='https://moztrap.mozilla.org/manage/case/add/' >+ New Case</Button>
+    } else if (this.props.setting.pagename == "suite") {
+      newButton = <Button bsStyle="success" href='https://moztrap.mozilla.org/manage/suite/add/' >+ New Suite</Button>
+    }
+    return (
+      <Row>
+        <Col md="12">
+          {newButton}
+          {diffButton}
+        </Col>
+      </Row>
+    )
+  }
+})
+
+
+var SearchableList = React.createClass({
+  mixins: [SearchableRemoteListMixin],
+
+  buildURL: function(query) {
+      var api_url =  config.baseUrl + "/api/v1/" + this.props.setting.pagename;
+      var codegen;
+      var pagename = this.props.setting.pagename;
+      if (pagename == "caseversion") {
+        codegen = caseversionCodegen;
+      } else if (pagename == "suite") {
+        codegen = suiteCodegen;
+      } else if (pagename == "caseselection") {
+        codegen = caseSelectionCodegen;
+      } else {
+        codegen = null;
+      }
+      return (buildQueryUrl(api_url, query, codegen) +
+              "&limit=" + config.defaultListLimit
+             );
+  },
+
+  handleQueueUpdate: function(e) {
+    var newState = {};
+    if (e.target.checked){
+      newState['checked'] = this.state['checked'].concat(e.target.value);
+    }
+    else {
+      this.state['checked'].splice(this.state['checked'].indexOf(e.target.value), 1);
+      newState['checked'] = this.state['checked'];
+    }
+    this.setState(newState);
+    console.log("===> handleQueueUpdate: ", newState);
+  },
+
+  render: function() {
+    return (
+      <div>
+        <ListToolbar setting={this.props.setting} checked={this.state.checked} />
+        <SearchForm ref="searchform" query={this.state.query} onSubmit={this.handleSearch} setting={this.props.setting}/>
+        <MTTable setting={this.props.setting} data={this.state.data} handleAddFilter={this.handleAddFilter} handleCheck={this.handleQueueUpdate}/>
+        <PaginationContainer onPageSelected={this.handlePageLoading} totalPageCount={this.state.queriedPageCount} />
+      </div>
+    )
+  }
+})
+
+// http://stackoverflow.com/questions/27864720/react-router-pass-props-to-handler-component
+var CaseVerWrapper = React.createClass({
```

Comment:
If we create a route like this
```
<Route name="resource" path="/:pagename" handler={SearchableList}/>
```
We can access the `pagename` variabe in `SearchableList` by `this.props.params.pagename`